### ISSUE TITLE
v0.31.2 Can't save Question for Empty Table

### ISSUE THREAD

**Issue by lindsay-stevens** [metabase/metabase#9286]
*Environment*
- Your browser and the version: Chrome 71
- Your operating system: Windows 10
- Your databases: Postgres
- Metabase version: 0.31.2
- Metabase hosting environment: Windows
- Metabase internal database: Postgres

*Repeatable steps to reproduce the issue*

- Create an empty table in the source database
- Run metadata sync
- Create query builder question based on empty table
- QB question fails to save, error:
  
```json
{"errors":
  {"result_metadata":
    "value may be nil, or if non-nil, value must be an array of valid results column metadata maps."
  }
}
```

- Add one or more rows to the table
- Create query builder question again
- QB question saves OK

**Comment by camsaul** [comment]
@lindsay-stevens are you still running into this issue in Metabase 0.32.8+? I can't reproduce it.

**Comment by camsaul** [comment]
Test to confirm this is fixed in #10166

**Comment by listentojohan** [comment]
I'm having this issue as well with the latest version.

**Comment by flamber** [comment]
@listentojohan 
So you're using version 0.33.2? And you see the same error in the Metabase log?
Does the sync process finish correctly or are you seeing other errors too?

Similar [report in the forum](https://discourse.metabase.com/t/question-not-saved-when-results-are-empty/7235) yesterday, but was using 0.32.8 and was recommended to upgrade to 0.32.10 and hasn't reported back that it didn't work.

**Comment by listentojohan** [comment]
@flamber I'm on 0.33.1
I'm not syncing the whole database/schema, that's simply too large.

**Comment by flamber** [comment]
@listentojohan 
1. Upgrade to 0.33.2 - that's the latest release - it fixed some breaking issues with 0.33.1
2. Are you see the same error in the Metabase log as reported here?
3. Do you see any other errors in the Metabase log?

**Comment by shinnawy** [comment]
Had this issue on 0.33.11, and can confirm that 0.33.2 fixed it. Just be sure to clear the cache.

**Comment by flamber** [comment]
Just to avoid any confusion about this is a returning issue, it's not.
The problem with specifically version 0.33.1 was that it could not save any questions - #10805 - and was fixed the same day and 0.33.2 was released.

**Comment by devmixture** [comment]
Clear your browser cache. Thanks @shinnawy
